sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,538 @@
+
+```ts
+{
+  type: 'some',
+  value: object // Clarity value
+}
+```
+
+`0x0b` `list`
+
+```ts
+{
+  type: 'list',
+  value: object[] // Array of Clarity values
+}
+```
+
+`0x0c` `tuple`
+
+```ts
+{
+  type: 'tuple',
+  value: Record<string, object> // Record of Clarity values
+}
+```
+
+`0x0d` `ascii`
+
+```ts
+{
+  type: 'ascii',
+  value: string // ASCII-compatible string
+}
+```
+
+`0x0e` `utf8`
+
+```ts
+{
+  type: 'utf8',
+  value: string
+}
+```
+
+#### Post-conditions
+
+`0x00` STX
+
+```ts
+{
+  type: 'stx-postcondition',
+  address: 'origin' | string | `${string}.${string}`, // Stacks c32-encoded, with optional contract name suffix
+  condition: 'eq' | 'gt' | 'gte' | 'lt' | 'lte',
+  amount: string // `bigint` compatible, amount in micro-STX
+}
+```
+
+`0x01` Fungible token
+
+```ts
+{
+  type: 'ft-postcondition',
+  address: 'origin' | string | `${string}.${string}`, // Stacks c32-encoded, with optional contract name suffix
+  condition: 'eq' | 'gt' | 'gte' | 'lt' | 'lte',
+  asset: `${string}.${string}::${string}` // Stacks c32-encoded address, with contract name suffix, with asset suffix
+  amount: string // `bigint` compatible, amount in lowest integer denomination of fungible token
+}
+```
+
+`0x02` Non-fungible token
+
+```ts
+{
+  type: 'nft-postcondition',
+  address: 'origin' | string | `${string}.${string}`, // Stacks c32-encoded, with optional contract name suffix
+  condition: 'sent' | 'not-sent',
+  asset: `${string}.${string}::${string}` // address with contract name suffix with asset suffix, Stacks c32-encoded
+  assetId: object, // Clarity value
+}
+```
+
+#### Test vectors
+
+Listed below are some examples of the potentially unclear representations:
+
+- `u12` = `{ type: "uint", value: "12" }`
+- `0xbeaf` = `{ type: "ascii", value: "hello there" }`
+- `"hello there"` = `{ type: "ascii", value: "hello there" }`
+- `(list 4 8)` =
+  ```
+  {
+    type: "list",
+    value: [
+      { type: "int", value: "4"},
+      { type: "int", value: "8"},
+    ]
+  }
+  ```
+- `(err u4)` =
+  ```
+  {
+    type: "err",
+    value: { type: "uint", value: "4"},
+  }
+  ```
+- "sends more than 10000 uSTX" =
+  ```
+  {
+    type: "stx-postcondition",
+    address: "STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6",
+    amount: "10000",
+    condition: "gt"
+  }
+  ```
+- "does not send the `12` TKN non-fungible token" =
+  ```
+  {
+    type: "ntf-postcondition",
+    address: "STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6.vault"
+    asset: "STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6.tokencoin::tkn",
+    assetId: { type: "uint", value: "12" }
+    condition: "not-sent"
+  }
+  ```
+
+### Provider registration
+
+Wallets can register their aliased provider objects however they see fit.
+For example, using the WBIP-004[^3] standard or Wallet Standard[^14].
+
+## Activation
+
+This SIP is considered _Ratified_ after Xverse and Leather (currently the largest wallets in the Stacks ecosystem) have implemented and launched the new standard.
```

Comment:
I'm not a fan of this phrasing - since it gives the impression that these changes are meant to benefit the two largest wallet providers. 

can it be rephrased so something more agnostic, like:
```
This SIP is considered _Ratified_ after at least two wallet providers have implemented and launched the new standard.
```
i'm not sure if it's possible to add some parameters here like "2 wallet providers with x number of downloads/users/dau etc", but that would help qualify it better.